packages/git-fetcher/src/index.ts

```diff
@@ -20,10 +20,10 @@ export default () => {
       const tempLocation = await cafs.tempDir()
       await execGit(['clone', resolution.repo, tempLocation])
       await execGit(['checkout', resolution.commit], { cwd: tempLocation })
-      await preparePackage(tempLocation)
       // removing /.git to make directory integrity calculation faster
       await rimraf(path.join(tempLocation, '.git'))
-      const filesIndex = await cafs.addFilesFromDir(tempLocation, opts.manifest)
+      await runPrepareHook(tempLocation)
+      const filesIndex = await filterFilesIndex(tempLocation, await cafs.addFilesFromDir(tempLocation, opts.manifest))
```

Comment:
but why do you add all files to the store? You're adding redundant files. Remove the redundant files from the directory before adding the directory to the store.

Reply:
Because I need to know what files are present that might not be part of the package. `packlist` is only gives me part of that: it tells me what files should be part of the package, but not what other files exist. I do think I probably know how to monkey patch it to return the list of files that exist though, at which point it would be possible to do what you suggest and only build cafs after the traversal. The code will be a bit icky though -- I'd be using undocumented aspects of the packlist implementation which could in theory break at any time unless the packlist version was fully locked.

Reply:
I'm not sure how to weigh the costs of doing an extra traversal to avoid computing unnecessary integrity hashes vs. the inflexibility of trying to integrate deeper into `packlist`. How important is perf to you? How much does computing those hashes cost? How much does a traversal cost? Without data I'd probably choose maintainability (that is, future correctness) over a perf upside. And at least this way there's only two traversals, which seem to me like less opportunity for things to become inconsistent than three would be. And perhaps eventually it would feasible to consolidate two down to one if a new implementation of this logic could be developed that was properly agnostic to the implementation of the base walker.

Reply:
ok, in this case it is better to walk the directory separately before running packlist. Yes, it will mean that the directory will be walked twice. But it is better than the alternative.

cafs is not only walking the directory. It is also calculating the hash of each file and adds each file to the store.